aiohttp/client_reqrep.py

```diff
@@ -428,10 +432,14 @@ def update_transfer_encoding(self) -> None:
             if hdrs.CONTENT_LENGTH not in self.headers:
                 self.headers[hdrs.CONTENT_LENGTH] = str(len(self.body))
 
-    def update_auth(self, auth: Optional[BasicAuth]) -> None:
+    def update_auth(self, auth: Optional[BasicAuth], trust_env: bool = False) -> None:
         """Set basic auth."""
         if auth is None:
             auth = self.auth
+        if auth is None and trust_env and self.url.host is not None:
```

Comment:
let's label the condition:
```suggestion
        URL_HAS_A_HOST = self.url.host is not None
        if auth is None and trust_env and URL_HAS_A_HOST:
```

Reply:
I don't remember if there are any tests for sockets, but it'd be interesting to check if we support HTTP queries to UNIX file sockets as well as abstract sockets. This is the only case I can think of not having a hostname. OTOH, theoretically `file://` URLs could fit this too.

Reply:
@webknjaz `aiohttp` client doesn't seem to support `file` URLs anyway. Regardless, I hope it's ok to leave this as is for now and not dig too much deeper.

Reply:
I realized that maybe URL for unix and abstract sockets would be another use-case. Although, we probably don't test if that works...
Anyway, let's keep it as is for now.